File: packages/python/palpluss/types/b2c.py
from __future__ import annotations

from typing_extensions import TypedDict


class B2cPayoutResponse(TypedDict):
    transactionId: str
    tenantId: str
    channelId: str | None
    type: str
    status: str
    amount: float
    currency: str
    phone: str
    reference: str | None
    description: str | None
    providerRequestId: str | None
    providerCheckoutId: str | None
    resultCode: str | None
    resultDescription: str
    idempotencyKey: str | None
    channel: None
    createdAt: str
    updatedAt: str

File: packages/typescript/src/types/b2c.ts
export interface B2cPayoutRequest {
  amount: number;
  phone: string;
  currency?: string;
  reference?: string;
  description?: string;
  channelId?: string;
  credential_id?: string;
  callback_url?: string;
}

export interface B2cPayoutResponse {
  transactionId: string;
  tenantId: string;
  channelId: string | null;
  type: 'B2C';
  status: 'PENDING';
  amount: number;
  currency: string;
  phone: string;
  reference: string | null;
  description: string | null;
  providerRequestId: string | null;
  providerCheckoutId: string | null;
  resultCode: string | null;
  resultDescription: string;
  idempotencyKey: string | null;
  channel: null;
  createdAt: string;
  updatedAt: string;
}
